gateways/adyen/adyen.go

```diff
@@ -50,19 +50,27 @@ func (client *AdyenClient) Authorize(request *sleet.AuthorizationRequest) (*slee
 	},
 	)
 
-	// potentially do something with http response
-	result, _, err := adyenClient.Checkout.Payments(buildAuthRequest(request, client.merchantAccount))
+	result, httpResp, err := adyenClient.Checkout.Payments(buildAuthRequest(request, client.merchantAccount))
+	var statusCode int
+	if httpResp != nil {
+		statusCode = httpResp.StatusCode
+	}
+	responseHeader := sleet.GetHTTPResponseHeader(request.Options, *httpResp)
```

Comment:
I think we use `request.Options` for other data in Adyen, is there some way to only get the options that are intended to be used to find response headers? Or would it make sense to pass in a function to get desired response headers?